### ISSUE TITLE
 login not anymore working

### ISSUE THREAD

**Issue by rocky-III** [YunoHost-Apps/ttrss_ynh#166]
yunohost.org -11.2.9.1 (stable)

Tiny Tiny RSS ttrss  20231202~ynh3
__

after update and a new installed Let's Encrypt-Certificate the is NO login anymore working... The installation was running for years now..

I tried the Default login credentials

    Username: admin
    password: password

and indeed i could login as admin and setup a new pw.... but all my feeds are organized in a different account...

 The credentials  do not work anymore by the web and also not by the android app. ... and i'm not able to set the pw back... i'm not 100% sure about the email i used for registration back than and tried to look the entry up in the phpmyadmin DB ttrss  /  ttrss_users  but can't find an email there for the user account... if i do an new entry in the table column for "email" this entry does not work and does not help to reset the pw ... 

What can i try next? Thanks for help.

**Comment by SimonMellerin** [comment]
Same here, 

I add that I'm a bit afraid because when I'm logged in with the user admin, I can't see my "real" user in the user panel in configuration ? 😱😱😱

Did I lost all my conf ?

**Comment by matthias4217** [comment]
I’ve updated today with the same result. It seems to be normal, as explained in the upgrade notes, TTRSS has migrated from MySQL to PGSQL, but the data could not be migrated. Everything is still present in the MySQL database though.

**Comment by SimonMellerin** [comment]
Damned, I should have been more attentive..

I'm gonna rollback then export my feeds to be able to reimport thème after.

Thanks

**Comment by zamentur** [comment]
SO to solve this, i discover a type incompatibility: 

My ttrss_feeds table was like this
```
CREATE TABLE public.ttrss_feeds (
 [...]
  last_updated timestamp without time zone,
  last_unconditional timestamp without time zone,
```

But the function was like this
```
create function SUBSTRING_FOR_DATE(timestamp, int, int) RETURNS text AS 'SELECT SUBSTRING(CAST($1 AS text), $2, $3)' LANGUAGE 'sql';
```
https://gitlab.tt-rss.org/tt-rss/tt-rss/-/blob/master/sql/pgsql/schema.sql?ref_type=heads#L337

I tried first to convert the columns to get timezone:
```
ALTER TABLE ttrss_feeds
  ALTER COLUMN last_updated
  SET DATA TYPE timestamp with timezone;
ALTER TABLE ttrss_feeds
  ALTER COLUMN last_unconditional
  SET DATA TYPE timestamp with timezone;
```

But it doesn't work cause `timestamp with timezone` and `timestamp` seems different !

I tried to specify only `timezone` but the timezoned columns are still here !
```
ALTER TABLE ttrss_feeds
  ALTER COLUMN last_updated
  SET DATA TYPE timestamp;
ALTER TABLE ttrss_feeds
  ALTER COLUMN last_unconditional
  SET DATA TYPE timestamp;
```

SO to fix this, i just made
```
create function SUBSTRING_FOR_DATE(timestamp with time zone, int, int) RETURNS text AS 'SELECT SUBSTRING(CAST($1 AS text), $2, $3)' LANGUAGE 'sql';
```


And it works 🎉 


So to sum up, i think to fix the upgrade we should add this:
```
pgloader --with "prefetch rows = 1000" mysql://$db_user:PASSWORD@127.0.0.1:3306/$db_name postgresql:///$db_user:PASSWORD@127.0.0.1:5432/$db_name
sudo -u postgres psql $db_name -c "create function SUBSTRING_FOR_DATE(timestamp without time zone, int, int) RETURNS text AS 'SELECT SUBSTRING(CAST($1 AS text), $2, $3)' LANGUAGE 'sql';"
```
